src/main/java/uk/gov/hmcts/cmc/claimstore/controllers/advices/ResourceExceptionHandler.java

```diff
@@ -233,4 +233,12 @@ public ResponseEntity<Object> handleUnprocessableEntity(Exception exception) {
             .status(UNPROCESSABLE_ENTITY)
             .body(new ExceptionForClient(UNPROCESSABLE_ENTITY.value(), exception.getMessage()));
     }
+
+    @ExceptionHandler(CoreCaseDataStoreException.class)
+    public ResponseEntity<String> handleCoreCaseDataException(CoreCaseDataStoreException exception) {
+        logger.error(exception);
+        return new ResponseEntity<>(exception.getMessage(),
+            new HttpHeaders(),
+            HttpStatus.INTERNAL_SERVER_ERROR);
```

Comment:
we don't want to throw 500 error here. more specific code will be good. How about `HttpStatus.FAILED_DEPENDENCY` ?

Reply:
Im taking it out.